qbft/spectest/tests/controller/latemsg/late_proposal_past_instance.go

```diff
@@ -36,31 +39,77 @@ func LateProposalPastInstance() tests.SpecTest {
 					height,
 				),
 			},
-
-			ControllerPostRoot: postRoot,
+			ControllerPostRoot:  sc.Root(),
+			ControllerPostState: sc.ExpectedState,
 		}
 	}
 
+	lateMsg := testingutils.TestingMultiSignerProposalMessageWithHeight([]*bls.SecretKey{ks.Shares[1]}, []types.OperatorID{1}, 2)
+	sc := lateProposalPastInstanceStateComparison(6, lateMsg)
+
 	return &tests.ControllerSpecTest{
 		Name: "late proposal past instance",
 		RunInstanceData: []*tests.RunInstanceData{
-			instanceData(qbft.FirstHeight, "24cf697092529cfab3ab06b969d8696692c8bcbb9f41a954f71dc74c3b1d7e97"),
-			instanceData(1, "8aa5464b119518f178d81edf4cea1f4c918f9e084e5262a0e276d3afb00ba620"),
-			instanceData(2, "1799fe0981ae08bde1eae9fef88ef8035f5952974647786287a1a8c36544a5da"),
-			instanceData(3, "2ddf2b8c2f35c8115ddd68120c71e64809bfea6b023ed13e177c2474a95d137d"),
-			instanceData(4, "318816cc8819ad062996704fb4b9990b8088ade0cb4c26816ea0965783bff12e"),
-			instanceData(5, "fd83cdee705de628cf4a9baf9e662c424bf0942c63a680266d7db872d16e9f0a"),
+			instanceData(qbft.FirstHeight),
+			instanceData(1),
+			instanceData(2),
+			instanceData(3),
+			instanceData(4),
+			instanceData(5),
 			{
 				InputValue: []byte{1, 2, 3, 4},
 				InputMessages: []*qbft.SignedMessage{
-					testingutils.TestingMultiSignerRoundChangeMessageWithHeight(
-						[]*bls.SecretKey{ks.Shares[1]},
-						[]types.OperatorID{1},
-						2,
-					),
+					lateMsg,
 				},
-				ControllerPostRoot: "db25cef2d243616e3d344cdb10f53875b4b06aad746a6fd476cab4c7c5dc6533",
+				ControllerPostRoot:  sc.Root(),
+				ControllerPostState: sc.ExpectedState,
 			},
 		},
+		ExpectedError: "could not process msg: invalid signed message: proposal is not valid with current state",
```

Comment:
why is this a new error? it didn't error before?

Reply:
Oleg fixed a bug in the test (we used roundchange msg before), so now we have this error message. Anyhow your comment made me think that maybe we can add another test. Added in #264